consensus/src/rand/rand_gen/rand_manager.rs

```diff
@@ -261,10 +268,14 @@ impl<
                         response_sender,
                     } = request;
                     match rand_gen_msg {
+                        RandMessage::RequestShare(request) => {
+                            if let Some(share) = self.rand_store.lock().get_self_share(request.rand_metadata()) {
```

Comment:
what will happen if we don't enter this `if`? Are we relying on the sender to timeout and retry?

Reply:
yes